app/logical/related_tag_calculator.rb

```diff
@@ -24,7 +25,7 @@ def self.frequent_tags_for_search(tag_query, search_sample_size: 1000, category:
     tags = tags.where("tags.post_count > 0")
     tags = tags.where(category: category) if category.present?
     tags = tags.order("overlap_count DESC, tags.post_count DESC, tags.name")
-    tags
+    [tags, sample_posts.length]
```

Comment:
did you mean `posts`?

in `self.frequent_tags_for_search` the variable is called `sample_posts` but it's passed to `self.frequent_tags_for_post_relation` as the value for the `posts` parameter

changing it to `posts` fixes it but - as there seems to be some kind of caching involved - changing it back didn't break it again *for the same search*, for a different one it would break again.

---

On my instance it causes errors such as this:
```
Error

undefined local variable or method `sample_posts' for RelatedTagCalculator:Module
Details

    ActionView::Template::Error exception raised
    app/logical/related_tag_calculator.rb:32:in `frequent_tags_for_post_relation'
    app/logical/related_tag_calculator.rb:21:in `frequent_tags_for_search'
    app/logical/related_tag_calculator.rb:9:in `similar_tags_for_search'
    app/logical/related_tag_calculator.rb:44:in `block (3 levels) in cached_similar_tags_for_search'
    app/logical/current_user.rb:93:in `without_safe_mode'
    app/logical/related_tag_calculator.rb:43:in `block (2 levels) in cached_similar_tags_for_search'
    app/models/application_record.rb:117:in `with_timeout'
    app/logical/related_tag_calculator.rb:42:in `block in cached_similar_tags_for_search'
    app/logical/cache.rb:18:in `get'
    app/logical/related_tag_calculator.rb:41:in `cached_similar_tags_for_search'
    app/presenters/post_set_presenters/post.rb:43:in `similar_tags'
    app/presenters/post_set_presenters/post.rb:28:in `related_tags'
    app/presenters/post_set_presenters/post.rb:13:in `tag_set_presenter'
    app/presenters/post_set_presenters/post.rb:59:in `tag_list_html'
    app/views/posts/index.html.erb:10
    app/views/posts/index.html.erb:1
    app/controllers/application_controller.rb:39:in `respond_with'
    app/controllers/posts_controller.rb:20:in `index'
```

Reply:
Yeah, I don't know how this happened, but there was a merge conflict with 9ddf408ec5152cc011a634cc8cde7f17e321669f, even though Github never reported a merge conflict when I submitted this 4 days ago.

Reply:
I'm removing the `sample_count` field. Calculating the sample count forces an extra query for all related tags calculations, even outside the API. The sample count should always be `min(search_count, 1000)`. For simple single tag searches, you can use the count of the largest tag in the `tags_overlap` field.

There was also a bug with the related tags button returning 250 tags instead of 25 during tag editing.

Reply:
We should probably return the post count data as well then. That should be easy enough, since the data is also there as well for free.

Reply:
Actually, nevermind, that may not be so useful. I'm thinking of an idea of including the post count for all of the displayed tags, but that would be better done as a separate query to the tags interface. Additionally, the post count of the tags wouldn't help for those times where the related tags query involves more than just simple tags, in which case a query to the post count endpoint would need to be done.